ONSdigital/dp-search-builder
# Add new health check to service

### What

Add new health endpoint with the following checkers:

* elasticsearch
* hierarchy api
* kafka consumer
* kafka producer
* kafka error producer (for reporter)

Re-worked entry point to service via main and renamed service package to event and moved graceful shutdown logic to run functionality to be consistent with other applications.

Also includes:
* Use new go dependency manage, modules instead of vendor
* Use latest libraries, dp-kafka, dp-rchttp, dp-api-clients-go, log.go, dp-elasticsearch

### How to review

* Check unit tests pass
* Run an integration test
* Check code changes make sense

### Who can review

Anyone
